tests/tr1/tests/cstdio/test.cpp

```diff
@@ -124,18 +126,17 @@ void test_cpp() { // test C++ header
     }
 
     { // test character I/O
-        const char* tmpbuff;
         char tname[L_tmpnam];
-        char tmpbuf[L_tmpnam];
+        char tn[100];
         STDx FILE* pf;
-        char tn[100] = {0};
 
-        assert(STDx tmpnam(tmpbuf) == tmpbuf);
-        CHECK(CSTD strlen(tmpbuf) < L_tmpnam);
-        assert((tmpbuff = STDx tmpnam((char*) nullptr)) != nullptr);
+        const auto temp_name1 = temp_file_name();
+        CHECK(temp_name1.size() < sizeof(tname));
+        CSTD strcpy_s(tname, sizeof(tname), temp_name1.c_str());
 
-        CSTD strcpy_s(tn, sizeof(tn), tmpbuff);
-        CSTD strcpy_s(tname, sizeof(tname), tmpbuf);
+        const auto temp_name2 = temp_file_name();

```

Comment:
I know its highly improbable but should we add a check that verifies  temp_name1 and temp_name2 are different? Same for other instances.

Reply:
Those checks already exist, for the cases where we simultaneously open 2 temp files. They are `tests/tr1/tests/cstdio/test.cpp` right line 141 and `tests/tr1/tests/cwchar1/test.cpp` right line 152:
```cpp
CHECK(CSTD strcmp(tn, tname) != 0);
```